ShellPkg/Library/UefiShellDriver1CommandsLib/DrvCfg.c

```diff
@@ -443,6 +469,18 @@ ConfigFromFile (
             // print out an error.
             //
             TempDevPathString = ConvertDevicePathToText ((EFI_DEVICE_PATH_PROTOCOL *)(((CHAR8 *)PackageHeader) + sizeof (EFI_HII_PACKAGE_HEADER)), TRUE, TRUE);
+            if (TempDevPathString == NULL) {
+              ShellPrintHiiEx (
+                -1,
+                -1,
+                NULL,
+                STRING_TOKEN (STR_GEN_OUT_MEM),
+                gShellDriver1HiiHandle,
+                L"drvcfg"
+                );
+              return (SHELL_OUT_OF_RESOURCES);

```

Comment:
I think `MainBuffer` (in the function) might need to be freed before returning.

Reply:
Agreed, added that and corrected the other returns in this function, some of which were not freeing `MainBuffer` and some of which were attempting to close the file handle after it had already been closed.